src/index.js

```diff
@@ -76,34 +101,32 @@ export class WebView extends Component {
     if (
       this.props.source.uri !== nextProps.source.uri ||
       this.props.source.method !== nextProps.source.method ||
-      this.props.source.body !== nextProps.source.body
+      this.props.source.body !== nextProps.source.body ||
+      this.props.source.html !== nextProps.source.html ||
+      this.props.source.baseUrl !== nextProps.source.baseUrl
     ) {
       this.handleSource(nextProps.source, nextProps.newWindow);
     }
   }
 
   componentWillUnmount() {
-    if (this.props.onMessage) {
-      window.removeEventListener('message', this.onMessage, true);
-    }
+    window.removeEventListener('message', this.onMessage, true);
   }
 
-  onMessage = (nativeEvent) => this.props.onMessage({ nativeEvent });
+  onMessage = (nativeEvent) => {
+    if (typeof this.props.onLoadEnd === 'function' && nativeEvent.data === 'DOMContentLoaded') {
+      this.props.onLoadEnd();
+    } else if (typeof this.props.onMessage === 'function') {
+      this.props.onMessage({ nativeEvent })
+    }
+  };
 
   postMessage = (message, origin) => {
     this.frameRef.contentWindow.postMessage(message, origin);
   };
 
-  handleInjectedJavaScript = (html) => {
-    if (this.props.injectedJavaScript) {
-      if (html) {
-        return html.replace('</body>', `<script>${this.props.injectedJavaScript}</script></body>`);
-      } else {
-        return html;
-      }
-    } else {
-      return html;
-    }
+  injectJavaScript = (expression) => {
+    this.frameRef.contentWindow.Function(`"use strict"; return ${expression};`)();
```

Comment:
This is similar to using `eval()`, but faster and safer.

Read "_Never use eval()!_" here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval